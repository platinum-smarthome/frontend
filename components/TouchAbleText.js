import React, { Component } from 'react'
import { Text, View, StyleSheet } from 'react-native'

class TouchAbleText extends Component {
  render() {
    return (
        <Text
          style={ styles.text }
          onPress={ this.props.onPress }
        >{this.props.text}</Text>
    )
  }
}

const styles = StyleSheet.create({
  text: {
    color: '#fff',
    opacity: 0.85,
    fontWeight: '200',
    fontSize: 14,
    borderBottomWidth: 1,
    borderBottomColor: '#fff',
  }
})

export default TouchAbleText