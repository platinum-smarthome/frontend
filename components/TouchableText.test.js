import React, { Component } from 'react'
import enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer'

import TouchAbleText from './TouchAbleText'
import { Text } from 'react-native'

enzyme.configure({ adapter: new Adapter()})

let wrapper

beforeEach(() => {
  wrapper = shallow(<TouchAbleText />)
})

describe('<TouchAbleText/>', () => {
  it('should render <TouchAbleText/>', () => {
    expect(wrapper).toBeDefined()
  })
})

describe('<TouchAbleText/> child is rendered', () => {
  it('Should render <Text/>', () => {
    expect(wrapper.find(<Text/>)).toBeDefined()
    expect(wrapper.find('Text').length).toBe(1)  
  })
})

describe('snapshot testing', () => {
  it('snapshot', () => {
    const tree = renderer.create(<TouchAbleText/>)
    expect(tree).toMatchSnapshot()
  })
})