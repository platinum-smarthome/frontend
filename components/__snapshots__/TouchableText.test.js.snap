// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing snapshot 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  onPress={undefined}
  style={
    Object {
      "borderBottomColor": "#fff",
      "borderBottomWidth": 1,
      "color": "#fff",
      "fontSize": 14,
      "fontWeight": "200",
      "opacity": 0.85,
    }
  }
/>
`;
